public/app/core/components/TransformersUI/ReduceTransformerEditor.tsx

```diff
@@ -1,42 +1,103 @@
-import React from 'react';
-import { StatsPicker } from '@grafana/ui';
+import React, { useCallback } from 'react';
+import { StatsPicker, Select, LegacyForms } from '@grafana/ui';
 import {
   DataTransformerID,
   ReducerID,
   standardTransformers,
   TransformerRegistyItem,
   TransformerUIProps,
+  SelectableValue,
 } from '@grafana/data';
 
-import { ReduceTransformerOptions } from '@grafana/data/src/transformations/transformers/reduce';
+import { ReduceTransformerOptions, ReduceTransformerMode } from '@grafana/data/src/transformations/transformers/reduce';
 import { selectors } from '@grafana/e2e-selectors';
 
 // TODO:  Minimal implementation, needs some <3
 export const ReduceTransformerEditor: React.FC<TransformerUIProps<ReduceTransformerOptions>> = ({
   options,
   onChange,
 }) => {
+  const modes: Array<SelectableValue<ReduceTransformerMode>> = [
+    {
+      label: 'Series to rows',
+      value: ReduceTransformerMode.SeriesToRows,
+      description: 'Create a table with one row for each series value',
+    },
+    {
+      label: 'Reduce fields',
+      value: ReduceTransformerMode.ReduceFields,
+      description: 'Collapse each field into a single value',
+    },
+  ];
+
+  const onSelectMode = useCallback(
+    (value: SelectableValue<ReduceTransformerMode>) => {
+      const mode = value.value!;
+      onChange({
+        ...options,
+        mode,
+        includeTimeField: mode === ReduceTransformerMode.ReduceFields ? !!options.includeTimeField : false,
+      });
+    },
+    [onChange, options]
+  );
+
+  const onToggleTime = useCallback(() => {
+    onChange({
+      ...options,
+      includeTimeField: !options.includeTimeField,
+    });
+  }, [onChange, options]);
```

Comment:
I don't understand how changes in this file relates to changes in live? Maybe open in a separate PR?

Reply:
dooh -- that is part of https://github.com/grafana/grafana/pull/27875

these three PRs are all required to make some good demos work ;)